### PR TITLE
Cryo Tweaks

### PR DESCRIPTION
**Pull request by bloxgate**
* Cryo cells no longer crush people to death
* Adds more cryoprotectant chemicals. Different levels of cold will need different levels of cryoprotection
* Adds the thermal regulator
  * Administers chemicals and heat to help maintain a normal **human** body temperature. (May be fatal if used on other species)
  * Also wakes people up if used after they're removed from cryo
* Gives the Bertels Deck 2 Medbay a thermal regulator.

### CODE REVIEW

File: code/modules/halo/chemicals/medicine.dm [hunk 1]
```diff
@@ -147,3 +146,71 @@
 		H.emote(pick("twitch", "blink_r", "shiver"))
 	H.add_chemical_effect(CE_SPEEDBOOST, 1)
 	H.add_chemical_effect(CE_PULSE, 2)
+
+/datum/reagent/cryoprethaline
```
Comment (X0-11) [inline code/modules/halo/chemicals/medicine.dm:150]: Don't these need recipes?

File: code/modules/halo/machinery/medical/thermopod.dm [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+	. = ..()
+	update_icon()
+
+/obj/machinery/thermopod/process()
+	if(stat & (NOPOWER|BROKEN))
+		return
+
+	if(occupant)
+		var/bodytemp = occupant.bodytemperature
+		if((bodytemp > 310 && bodytemp < (310 + 40 * TEMPERATURE_DAMAGE_COEFFICIENT)) || (bodytemp < 311 && bodytemp > (311 - 40 * TEMPERATURE_DAMAGE_COEFFICIENT))) //Would leporazine work fast?
```
Comment (X0-11) [inline code/modules/halo/machinery/medical/thermopod.dm]: Couldn't this compare to the occupant's species settings for bodytemps?

File: code/modules/halo/machinery/medical/thermopod.dm [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+		visible_message("\The [user] starts climbing into \the [src].")
+	else
+		visible_message("\The [user] starts putting [M] into \the [src].")
+
+	if(do_after(user, 20, src))
+		if(occupant)
+			to_chat(user, "<span class='warning'>\The [src] is already occupied.</span>")
+			return
+		M.stop_pulling()
+		if(M.client)
```
Comment (X0-11) [inline code/modules/halo/machinery/medical/thermopod.dm:86]: I think M.reset_view() would do what you're looking for here

File: code/modules/halo/machinery/medical/thermopod.dm [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+		M.forceMove(src)
+		update_use_power(2)
+		occupant = M
+		update_icon()
+
+//Rip from sleeper code
+/obj/machinery/thermopod/proc/go_out()
+	if(!occupant)
+		return
+	if(occupant.client)
```
Comment (X0-11) [inline code/modules/halo/machinery/medical/thermopod.dm:98]: Same here.

File: maps/first_contact/maps/UNSC_Bertels/UNSC_Bertels_Deck_2.dmm [hunk 5]
```diff
@@ -167,6 +167,7 @@
 "dk" = (/obj/structure/bed/chair{dir = 4},/obj/machinery/light{dir = 8; icon_state = "tube1"; pixel_y = 0},/turf/simulated/floor/tiled,/area/corvette/unscbertels/deck2/marinemesshall)
 "dl" = (/obj/structure/table/standard,/turf/simulated/floor/tiled,/area/corvette/unscbertels/deck2/marinemesshall)
 "dm" = (/obj/structure/bed/chair{tag = "icon-chair_preview (WEST)"; icon_state = "chair_preview"; dir = 8},/turf/simulated/floor/tiled,/area/corvette/unscbertels/deck2/marinemesshall)
+"dn" = (/obj/machinery/thermopod,/turf/simulated/floor/tiled/white,/area/corvette/unscbertels/deck2/starboardbay)
```
Comment (X0-11) [inline maps/first_contact/maps/UNSC_Bertels/UNSC_Bertels_Deck_2.dmm:170]: Innies might need one on the thorn too.

**Review by X0-11 — DISMISSED** [review]

File: code/modules/halo/machinery/medical/thermopod.dm (hunk 3 above)
Comment (bloxgate) [reply code/modules/halo/machinery/medical/thermopod.dm:86]: This is a straight rip from the sleeper code. I can change it if we want, but this is proven to work.

Comment (X0-11) [reply code/modules/halo/machinery/medical/thermopod.dm:86]: For the sake of consistency and knowing it works, let's keep it like this then.

File: code/modules/halo/machinery/medical/thermopod.dm (hunk 2 above)
Comment (bloxgate) [reply code/modules/halo/machinery/medical/thermopod.dm]: That could be done, but we'd have to then change leporazine to check that as well. That check is checking if their body temperature is in the 1 tick treatment range of leporazine

File: maps/first_contact/maps/UNSC_Bertels/UNSC_Bertels_Deck_2.dmm (hunk 5 above)
Comment (bloxgate) [reply maps/first_contact/maps/UNSC_Bertels/UNSC_Bertels_Deck_2.dmm:170]: Will do. Is there anywhere else that you can think of?

File: code/modules/halo/chemicals/medicine.dm (hunk 1 above)
Comment (bloxgate) [reply code/modules/halo/chemicals/medicine.dm:150]: I can add some. Any suggestions for ingredients? Besides glycerol for the two cryo ones?

Comment (X0-11) [reply code/modules/halo/chemicals/medicine.dm:150]: You could probably make the higher strength chems need some of the lower strength ones to be made

**Comment by BDpuffy420** [comment]
file conflict

**Comment by bloxgate** [comment]
File conflict fixed, and requested changes made.

**Comment by BDpuffy420** [comment]
Still needs changelog

**Review by BDpuffy420 — APPROVED** [review]

File: code/modules/halo/machinery/medical/thermopod.dm [hunk 6]
```diff
@@ -12,7 +12,7 @@
 	anchored = 1
 	clicksound = 'sound/machines/buttonbeep.ogg'
 	clickvol = 30
-	var/mob/living/carbon/human/occupant = null
+	var/tmp/mob/living/carbon/human/occupant = null
```
Comment (caelaislinn) [inline code/modules/halo/machinery/medical/thermopod.dm:15]: tmp vars only affect the savefile, which thermopod doesn't have anything to do with?

Comment (bloxgate) [reply code/modules/halo/machinery/medical/thermopod.dm:15]: I heard some discussion about using tmp vars for things like this makes mapping work faster. Though, I probably just horribly misinterpreted that.

Either way I don't think it should make any difference, unless we end up saving an in game map again.

File: code/modules/halo/chemicals/medicine.dm [hunk 7]
```diff
@@ -1,4 +1,3 @@
-
```
Comment (caelaislinn) [inline code/modules/halo/chemicals/medicine.dm:1]: Grr unnecessary line change

File: code/modules/mob/living/carbon/human/life.dm [hunk 8]
```diff
@@ -390,27 +390,31 @@
...
+			if(chem_effects[CE_CRYO] < needed_cryo)
 				take_overall_damage(burn=burn_dam, used_weapon = "Low Body Temperature")
 				fire_alert = max(fire_alert, 1)
 
 	// Account for massive pressure differences.  Done by Polymorph
 	// Made it possible to actually have something that can protect against high pressure... Done by Errorage. Polymorph now has an axe sticking from his head for his previous hardcoded nonsense!
 	if(status_flags & GODMODE)	return 1	//godmode
 
-	if(adjusted_pressure >= species.hazard_high_pressure)
+	//Make people not take  highpressure damage in cryo. Bloxgate - 2019-05-18
```
Comment (caelaislinn) [inline code/modules/mob/living/carbon/human/life.dm:412]: See you in another 10 years! -CA

**Review by caelaislinn — COMMENTED** [review]
I'm still a bit confused. Why is this needed? Internal body temp is usually only a symptom of other medical problems

File: code/modules/halo/chemicals/medicine.dm (hunk 7 above)
Comment (bloxgate) [reply code/modules/halo/chemicals/medicine.dm:1]: That was removing a random blank line at the top of the file. I can put it back if you want?

Comment (caelaislinn) [reply code/modules/halo/chemicals/medicine.dm:1]: In the past myself and others have done that for code readability. Doesn't matter though just very slightly bloats code review

**Comment by bloxgate** [comment]
This is mainly intended to make waking people up after cryo faster, just stick themselves in and they get let themselves out after a second or two rather than having to shake them back awake or wait for them to wake up normally. Also has a few niche uses, if someone somehow manages to get their body temperature dangerously high or low.

The chemicals themselves also have a few possible usages. You could give people small doses of hexaline or something if we ever get hazardously cold planets or something.

Edit: Also the cells really shouldn't crush people if their loops are built too small by mappers.

**Review by caelaislinn — APPROVED** [review]

**Comment by caelaislinn** [comment]
Sounds good

**Pull request merged**